src/lib/HST/Effect/InputFile.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | This module defines an effect for computations that can store the file
+--   path and the content of an input file.
+module HST.Effect.InputFile
+  ( -- * Effect
+    InputFile
+    -- * Actions
+  , setFile
+  , getFile
+    -- * Interpretations
+  , runInputFile
+  ) where
+
+import           Polysemy       ( Sem, makeSem, reinterpret )
+import           Polysemy.State ( State, evalState, get, put )
+
+-------------------------------------------------------------------------------
+-- Effect and Actions                                                        --
+-------------------------------------------------------------------------------
+-- | An effect capable of setting and getting the file path and the content of
+--   the input file.
+--
+--   While setting the input file, the content is converted to lines of code.
+data InputFile m a where
+  SetFile :: FilePath -> String -> InputFile m ()
+  GetFile :: InputFile m (FilePath, [String])
```

Comment:
If the file name is added to the source span, it would make sense to generalize this effect such that there can be multiple input files.

```suggestion
  GetFile :: FilePath -> InputFile m [String]
```

This would mean that a map has to be stored by `runWithInputFile` but would provide a cleaner default value for `runInputFile`.